second commit

LikeARivers opened the pull request:
добавлены тесты и отчеты

LikeARivers committed:
second commit

irisqul commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,40 @@
+    def test_two_create_courier(self, two_courier):
+        created_two_courier_request = two_courier
+
+        assert created_two_courier_request.status_code == 409 and created_two_courier_request.json()["message"] == "Этот логин уже используется"
```
Нужно исправить: тест падает, хотя не должен

irisqul commented on conftest.py:
```diff
@@ -0,0 +1,84 @@
+import urls
+
+
+@allure.step("СОЗДАНИЕ, авторизация, удаление рандомного курьера курьера")
```
Нужно исправить: фикстуры не универсальны, содержат дублирование кода. Нам нужно собирать код как конструктор - сделать отдельно фикстуру генерации данных, отдельно - регистрации, отдельно - авторизации - каждая следующая используется предыдущую. В тестах вызываем те, что нужны.

irisqul commented on conftest.py:
```diff
@@ -0,0 +1,84 @@
+    courier_id = auth_courier_response.json().get("id")
+    delete_courier_response = requests.delete(f"{urls.BASE_URL}{urls.DELETE_COURIER_ENDPOINT}{courier_id}")
+
+@allure.step("Создание, АВТОРИЗАЦИЯ с пустым паролем, удаление курьера")
```
Нужно исправить: валидация полей - это шаги теста, а не фикстуры. мы не тестируем фикстуры, поэтому часть написаного в фикстурах нужно удалить и совершать в шагах тестов.

irisqul commented on tests/test_auth_courier.py:
```diff
@@ -0,0 +1,31 @@
+    @allure.title("Проверка успешной авторизации курьера")
+    @allure.description("Авторизация рандомно созданного курьера с логином и паролем, проверка статуса и тела ответа")
+    def test_success_auth_random_courier(self, auth_random_courier):
+        response_courier = auth_random_courier
```
Нужно исправить: в этом и многих тестах - нет шагов. Мы не тестируем фикстуры, фикстура - это состояние системы до начала теста, а не замена теста

LikeARivers committed:
фикстуры исправлены на универсальные

шаги перенесены в тест

валидация перенесена в тест

про ошибку создания двух одинаковых курьеров написано в README
